extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +152,208 @@ export class Transport {
+      frameData = Transport.getFrameData(frameUrl);
+      ++(frameData.usageCount);
+    } else {
+      frameData = this.createCrossDomainIframe(win, frameUrl);
+      shouldAddToDOM = true;
+    }
+    dev().assert(frameData, 'Trying to use non-existent frame');
+    if (opt_processResponse) {
+      Transport.responseProcessors_[this.id_] = opt_processResponse;
+    }
+    if (shouldAddToDOM) {
+      win.document.body.appendChild(frameData.frame);
+    }
+  }
+
+  /**
+   * Create a cross-domain iframe for third-party vendor anaytlics
+   * @param {!Window} win The window element
+   * @param {!string} frameUrl  The URL of the cross-domain iframe
+   * @param {function(!string,
+   *   !../../../src/3p-analytics-common.AmpAnalytics3pResponse)=}
+   *   opt_processResponse An optional function to receive any response
+   *   messages back from the cross-domain iframe
+   * @return {!FrameData}
+   * @VisibleForTesting
+   */
+  createCrossDomainIframe(win, frameUrl, opt_processResponse) {
+    const sentinel = Transport.createUniqueId_();
+    const useLocal = getMode().localDev || getMode().test;
+    const useRtvVersion = !useLocal;
+    const scriptSrc = calculateEntryPointScriptUrl(
+        window.location, 'ampanalytics-lib', useLocal, useRtvVersion);
+    const frameName = JSON.stringify(/** @type {JsonObject} */ ({
+      scriptSrc,
+      sentinel,
+    }));
+    const frame = createElementWithAttributes(win.document, 'iframe',
+        /** @type {!JsonObject} */ ({
+          sandbox: 'allow-scripts allow-same-origin',
+          name: frameName,
+          src: frameUrl,
+          'data-amp-3p-sentinel': sentinel,
+        }));
+    frame.sentinel = sentinel;
+    setStyles(frame, {
+      width: 0,
+      height: 0,
+      display: 'none',
+      position: 'absolute',
+      top: 0,
+      left: 0,
+    });
+    const frameData = /** @const {FrameData} */ ({
+      frame,
+      usageCount: 1,
+      queue: new AmpIframeTransportMessageQueue(win,
+          /** @type {!HTMLIFrameElement} */
+          (frame)),
+    });
+    Transport.crossDomainIframes_[frameUrl] = frameData;
+    // Note: this is a listener for responses from the frame, regardless of
+    // which transport instance a response is intended for. This is because
+    // the listener may only be created prior to the iframe being attached
+    // to the DOM - see assert in listenFor which references #2942.
+    // This is why listenFor() is called here in createCrossDomainIframe(),
+    // rather than in processCrossDomainIframe().
+    frameData.responseMessageUnlisten = listenFor(frameData.frame,
+        AMP_ANALYTICS_3P_MESSAGE_TYPE.RESPONSE,
+        response => {
+          dev().assert(response && response['data'],
+              'Received empty response from 3p analytics frame');
+          const responseProcessor = Transport.responseProcessors_[
+            response['transport']];
+          if (responseProcessor) {
```

Comment:
Note that at this point, all we have is the ID of the intended recipient. Do you want me to build an ID->URL (or ID->transport instance) map/getter just to make this log message be able to inform about the URL?